[워터]  학습 계획 정리

yesolyo opened the pull request.

yesolyo committed:
feat: sidebar 레이아웃 완료

기존 sidebar.html을 header.html에 합치고 닫기 레이아웃도 포함하여
제작완료하였습니다.

yesolyo committed:
feat: login 팝업창을 hover했을 경우 배경 dimmed 처리 완료

login 팝업창을 hover했을 때 배경을 어둡게 dimmed 처리를 하고, hover하지
않을때는 dimmed 처리가 되지 않게 처리하였습니다.

yesolyo committed:
docs: 1주차 회고와 2주차 학습 계획 정리

1주차를 진행하면서 느꼈던 점과 개선 점을 정리하였으며, 2주차 학습계획을
일주일로 나눠서 계획을 작성하였습니다.

yesolyo committed:
fix: login 팝업창과 배송처 팝업 창 hover 시 header 제외 dimmed 처리

기존에 header도 포함하여 dimmed처리가 되었으나 header를 제외하여
dimmed 처리로 변경하였습니다.(로그인 팝업창과 배송처 팝업창 hover시)

yesolyo committed:
feat: sub menu에서 모두 클릭시 sidebar 나타나게 설정 완료

sub menu를 클릭하였을때 sidebar가 화면에 나타나게 설정하였습니다.

yesolyo committed:
feat: 사이드바에서 x를 클릭 시 sidebar가 닫히는 기능 완료

사이드바에서 x를 클릭 시 sidebar가 닫혀지는 기능을 완료하였습니다.

yesolyo committed:
feat: sub menu의 모두 클릭시 sidebar가 나오면서 배경 딤처리 완료

sidebar가 나오면 header부분을 포함하여 모든 배경이 딤처리가 되게
완료하였습니다.

yesolyo committed:
feat: 사이드 바가 화면에 나올때 로그인 팝업창도 사라지게 설정 완료

sub menu에서 모두를 클릭시 sidebar가 나올때 로그인 팝업창도 같이
사라지게 설정을 완료하였습니다.

crongro reviewed (commented):
2주차 항목도 엄청 많네요.
요일별 계획까지 꼼꼼하게 좋네요.

yesolyo committed:
refactor: 필요없는 태그 삭제

하이퍼링크 태그가 필요가 없어서 삭제하였습니다.

yesolyo committed:
feat: 사이드바 hover시 배경화면 색깔과 아이콘 색깔 변경

사이드바 hover시 배경화면 색깔과 아이콘 색깔이 변경되게 완료하였습니다.

yesolyo committed:
fix: sidebar close하고 배송처를 클릭시 전체 화면 딤처리되는 현상 해결

sidebar close버튼을 클릭 후 배송처 hover시 전체 화면 딤처리가 되어서
배송처를 제외한 배경화면을 딤처리하였습니다.

yesolyo committed:
feat: sidebar 모두보기 클릭시 detail menu 확인 기능

sidebar에서 모두보기를 클릭했을 경우 detail menu list를 확인 할 수
있으며, detail menu list에서 간단히 보기를 클릭 했을 경우 detail menu
list가 close되게 하였습니다.

yesolyo committed:
refactor: sidebar에 들어가는 data 정리 파일 생성

sidebar에 들어가는 데이터를 가져오기 위해서 data가 들어있는 파일을
정리하였습니다.

crongro commented on css/sidebar.css:
```diff
@@ -26,13 +37,20 @@
+}
+.sidebar__all-menu_extend > li:hover img,
+.sidebar__menu > ul > li:hover img{
+    filter: invert(85%) sepia(58%) saturate(4611%) hue-rotate(317deg) brightness(102%) contrast(95%);
```
뭔지 모르지만 복잡해보이는군요 ㅎ

crongro commented on js/dimmed.js:
```diff
@@ -0,0 +1,37 @@
+  dimmed.style.display = "none";
+});
+
+shippingMenu.addEventListener("mouseover", () => {
+  dimmed.style.display = "block";
+});
```
이 세 줄을 한 줄로 바꾸보세요. { } 제거하고요.

crongro commented on js/sidebar_data.js:
```diff
@@ -0,0 +1,26 @@
+menuAll.addEventListener("click", () => {
+  if (!menuAllDetail.innerHTML) {
+    TITLE_DETAIL.forEach((data) => {
+      menuAllDetail.insertAdjacentHTML(
```
insertAdjacentHTML 사용하신분을 여기서 발견하네요.

crongro commented on main.html:
```diff
@@ -0,0 +1,195 @@
+                            </div>
+                            <div class="login_popup_extension-right-box">
+                                <ul>
+                                    <li class="login_popup_extension-title">계정</li>
+                                    <li class="login_popup_extension-list">계정</li>
+                                    <li class="login_popup_extension-list">주문</li>
+                                    <li class="login_popup_extension-list">권장 사항</li>
+                                    <li class="login_popup_extension-list">검색 기록</li>
+                                    <li class="login_popup_extension-list">위치리스트</li>
+                                    <li class="login_popup_extension-list">비디오 구매 및 대여</li>
+                                    <li class="login_popup_extension-list">Kindle 언리미티드</li>
+                                    <li class="login_popup_extension-list">콘테츠 및 기기</li>
+                                    <li class="login_popup_extension-list">항목 구독 및 저장</li>
+                                    <li class="login_popup_extension-list">멤버십 및 구독</li>
+                                    <li class="login_popup_extension-list">음악 라이브러리</li>
```
이런형태도 배열+반복문으로 화면에 동적으로 추가할 수 있을듯하군요.

crongro reviewed (commented):
dimmed.js 는 어떤것을 dimmed 하는것인지 구체적인 이름으로 바꾸셔도 좋고요.
함수를 사용해서 전역변수를 모두 없애보세요.
